File: jar-common/src/main/java/com/beancounter/common/input/PortfolioInput.java
package com.beancounter.common.input;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.RequiredArgsConstructor;

@Data
@Builder
@RequiredArgsConstructor
@AllArgsConstructor
public class PortfolioInput {
  private String code;
  private String name;
  private String currency;
  @Builder.Default
  private String base = "USD";
}

File: jar-common/src/main/java/com/beancounter/common/input/PortfolioInput.kt
package com.beancounter.common.input

/**
 * Basic portfolio request contract. These are tied to the Caller and do not accept a primary key.
 * The Code is expected to be unique to the caller.
 */
data class PortfolioInput(val code: String, var name: String, var currency: String, var base: String) {
    constructor(code: String, name: String, currency: String) : this(code, name, currency, "USD")
}
